browser/src/app/TilesMiddleware.ts

```diff
@@ -1373,13 +1376,21 @@ class TileManager {
 		return boundList.map(this.pxBoundsToTileRange, this);
 	}
 
-	private static getMissingTiles(pixelBounds: any, zoom: number) {
+	private static getMissingTiles(
+		pixelBounds: any,
+		zoom: number,
+		updateCurrent: boolean,
+	) {
 		var tileRanges = this.pxBoundsToTileRanges(pixelBounds);
 		var queue = [];
 
+		if (updateCurrent) {
+			for (var key in this.tiles) this.tiles[key].current = false;
```

Comment:
I'm suspicious of this. We can have multiple tile ranges that are current in split pane cases I would expect (?) =) So we need dis-contiguous regions of live / current tiles all rendered on the screen.

Otherwise this looks overall great - we need Gokay's review of his bits =)

Reply:
Yes, I suspect this is broken too - I wonder if we can just get rid of current and calculate it as necessary instead - that would really reduce complexity a lot... I suppose it's only really needed for garbage collection and that doesn't happen so frequently that we need to cache this value I don't think(?)

Reply:
We could try calculating it on-demand; but keeping that in-sync is an issue - and we shouldn't need to be perfect. One thing we could do that is simpler is just sort by the last-rendered timestamp (I guess) and free the oldest - presumably dumping a 'now' into a member is fast and easy as we whack the canvas on screen. Its not clear to me that we can safely assume we have all the canvases we need in-memory to render to the screen anyway.

Reply:
Turns out it isn't broken - `getMissingTiles` is called with a bounds list, not a rect - so split panes work fine with this patch :) I think it would be nice to get rid of `current` and I think it'd be a relatively small change on top of this, but perhaps one thing at a time. I'm going to start splitting this up - test results also looking encouragingly positive.

Reply:
If you're happy lets get it in =)

Reply:
I wouldn't be surprised if there are edge-cases where this does the wrong thing - I haven't hit one yet (been messing about in writer/calc/impress, and with split panes), but I definitely don't know how to exercise this code as well as others :) - All the more reason to get rid of Tile.current though, which I'm currently (hah) looking at. It's a bigger change than I thought, so definitely subsequent work, but it should significantly simplify what's happening here.